src/Compilers/Core/RebuildTest/RoundTripUtil.cs

```diff
@@ -41,16 +43,31 @@ internal static class RoundTripUtil
 
             using var peReader = new PEReader(peStream);
             var optionsReader = new CompilationOptionsReader(logger, peReader.GetEmbeddedPdbMetadataReader(), peReader);
+            VerifyMetadataReferenceInfo(optionsReader, metadataReferences);
+
             var compilationFactory = CompilationFactory.Create(
                 assemblyFileName,
                 optionsReader);
             using var rebuildPeStream = new MemoryStream();
             var emitResult = compilationFactory.Emit(rebuildPeStream, syntaxTrees, metadataReferences, cancellationToken);
             Assert.True(emitResult.Success);
-
             Assert.True(peStream.ToArray().SequenceEqual(rebuildPeStream.ToArray()));
         }
 
+        public static void VerifyMetadataReferenceInfo(CompilationOptionsReader optionsReader, ImmutableArray<MetadataReference> metadataReferences)
+        {
+            var count = 0;
+            foreach (var info in optionsReader.GetMetadataReferences())
+            {
+                count++;
+                var metadataReference = metadataReferences.FirstOrDefault(x =>
+                    info.Mvid == x.GetModuleVersionId() &&
+                    info.ExternAliases.SequenceEqual(x.Properties.Aliases));
+                AssertEx.NotNull(metadataReference);
```

Comment:
>AssertEx [](start = 16, length = 8)

`Assert`?

Reply:
The standard `Assert` type isn't nullable aware and hence won't set the state to non-null.